system/security/cryptohelper/cryptocommon.cpp

```diff
@@ -76,26 +76,31 @@ MODULE_EXIT()
 namespace cryptohelper
 {
 
-static void populateErrorFormat(StringBuffer &formatMessage, const char *format)
+static void populateError(StringBuffer &msg)
 {
     // openssl doesn't define max error string size, but 1K will do
-    char *evpErr = formatMessage.reserve(1024);
+    char *evpErr = msg.reserve(1024);
     ERR_error_string_n(ERR_get_error(), evpErr, 1024);
-    formatMessage.setLength(strlen(evpErr));
-    formatMessage.append(" - ").append(format);
+    msg.setLength(strlen(evpErr));
+    msg.append(" - ");
 }
+
+static IException *makeEVPExceptionVA(int code, const char *format, va_list args)  __attribute__((format(printf,2,0)));
```

Comment:
does this need to be forward declaration?
i.e. couldn't you add static and the printf format __attribute to the line below?

Reply:
No, compiler does not allow that (I don't know why not)